File: app/lib/models/reward.dart
class Reward {
  final String id;
  final String name;
  final String? description;
  final int cost;
  final int? stock;
  final int redeemed;
  final String? category;
  final int? limitPerParticipant;
  final bool available;

  const Reward({
    required this.id,
    required this.name,
    this.description,
    required this.cost,
    this.stock,
    required this.redeemed,
    this.category,
    this.limitPerParticipant,
    required this.available,
  });

  factory Reward.fromJson(Map<String, dynamic> json) => Reward(
        id: json['id'] as String,
        name: json['name'] as String,
        description: json['description'] as String?,
        cost: json['cost'] as int,
        stock: json['stock'] as int?,
        redeemed: json['redeemed'] as int? ?? 0,
        category: json['category'] as String?,
        limitPerParticipant: json['limitPerParticipant'] as int?,
        available: json['available'] as bool? ?? true,
      );

  int? get remaining => stock != null ? stock! - redeemed : null;
  bool get inStock => stock == null || remaining! > 0;
}

File: backend/src/entities/Reward.ts
import {
  Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn,
  ManyToOne, OneToMany, JoinColumn,
} from 'typeorm';
import { Event } from './Event.js';
import { Redemption } from './Redemption.js';

@Entity('rewards')
export class Reward {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ length: 200 })
  name: string;

  @Column({ type: 'text', nullable: true })
  description: string | null;

  @Column({ type: 'int' })
  cost: number;

  @Column({ type: 'int', nullable: true })
  stock: number | null;

  @Column({ type: 'int', default: 0 })
  redeemed: number;

  @Column({ length: 100, nullable: true })
  category: string | null;

  @Column({ type: 'int', nullable: true })
  limitPerParticipant: number | null;

  @Column({ default: true })
  available: boolean;

  @Column({ type: 'int', default: 0 })
  sortOrder: number;

  @ManyToOne(() => Event, (e) => e.rewards, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'eventId' })
  event: Event;

  @Column('uuid')
  eventId: string;

  @OneToMany(() => Redemption, (r) => r.reward)
  redemptions: Redemption[];

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}
